HOF фильтр

FryOS opened the pull request.

FryOS committed:
master folder

FryOS committed:
HOF фильтер

**What:**

**Why:**
-Fix #xxx

**How:**
-

korbutds commented on 10-array-clean/app.js:
```diff
@@ -0,0 +1,26 @@
+    for (let i = 0; i < arr.length; i++) {
+        const element = arr[i];
+        if(removeItem(element)){
+            filterArray.push(element);
```
Если `Удалить элемент`, то `Добавить элемент`?

FryOS replied:
Переделал имя

korbutds replied:
Так а что изменилось. Ваша функция, судя по названию, спрашивает: “Нужно ли удалить элемент”. И если тру, то вы его добавляете в результирующий массив. Тут нужно пересмотреть функцию. Логичнее было что-то типа !removeEl или includeEl. Что-то типа того.

FryOS replied:
Не понимаю, проблема в том что функция не рабочая или в имени? isRemoved переводится как удалена ли? Или Удалена? Ложь или правда

korbutds replied:
Либо в имени, либо в функциональности. Попробуйте прочитать условие. Если (удалить функция (элемент)), то  добавить элемент в результрующий массив. Тут логичнее было бы:
```js
if (!removeFn(element)) {
 filterArray.push(element)
} 

// or

if (filterFn(element)) {
  filterArray.push(element)
}
```

Стало понятнее?

FryOS replied:
не понимаю, а что не так? вот есть функция removeFn она возвращает ложь или правду. я ее использую. вот как и все. не понимаю что не так?
разница в коде только что есть ! или его нет, а что это дает? можно развернутый ответ ?

korbutds replied:
У Вас функция называется, `ФункцияУдаления`. Судя по названию она должна возвращать `true` если элемент нужно удалить, а не добавить.

FryOS replied:
Ну а как надо? я запутался. давайте так, сам код нормальный? он решает то что надо в задании? по мне так да . 

А что надо исправить и на что?

korbutds replied:
Переименуйте параметр с `removeFn` на `filterFn`

korbutds replied:
Еще раз подытожу:
1. Есть функция `isRemovedItem`, которая как бы говорит, если соблюдается условие, то верни `true`. Т.е. судя по названию она должна удалять, т.е. не добавлять элемент в результирующий массив. 
2. Есть условие, которое использует эту функцию. И сейчас вы используете его значение. Т.е. смотрите комментарии выше. 
3. В результате вы возвращаете массив протиоположного условия. Т.е. вроде как нужно было отфильтровать элементы которые меньше 5, а у Вас все наоборот.

FryOS replied:
переименовал

korbutds replied:
Где?

FryOS replied:
Сделал.

korbutds commented on 10-array-clean/app.js:
```diff
@@ -0,0 +1,26 @@
+    return filterArray;
+}
+
+
```
Лишний пробел

FryOS replied:
Убрал

FryOS committed:
#xxx: Commit title

Переделал функцию удаления и имя функции проверки
**What:**

**Why:**
-Fix #xxx

**How:**
-

FryOS committed:
переименовал функцию, фильтрация < 5
